extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -14,29 +19,63 @@ import {
  */
 let ShoppingConfigResponseDef;
 
+/**
+ * Uses the specified validation configuration to run validation against
+ * the user's shopping configuration.
+ * @param {!ShoppingConfigDataDef} productConfig The user's config object.
+ * @param {string} name the name of the item in the config object.
+ * @return {boolean} returns a boolean indicating whether the validation
+ *     was successful.
+ */
+function validateConfig(productConfig, name) {
+  const errors = validateProduct(productConfig, name);
+  if (errors.length > 0) {
+    user().warn('AMP-STORY-SHOPPING-CONFIG', `${errors}`);
+    return false;
+  }
+  return true;
+}
+
 /** @typedef {!Object<string, !ShoppingConfigDataDef> */
-export let KeyedShoppingConfigDef;
+let KeyedShoppingConfigDef;
 
 /**
- * Gets Shopping config from an element.
- * The config is validated and keyed by 'product-tag-id'.
- * @param {!Element} element <amp-story-shopping-attachment>
- * @return {!Promise<!KeyedShoppingConfigDef>}
+ * Validates and returns the shopping config corresponding to the given
+ * amp-story-shopping-attachment element.
+ * @param {!Element} shoppingAttachmentEl <amp-story-shopping-attachment>
+ *     The amp story shopping attachment element
+ * @return {!Promise<!KeyedShoppingConfigDef>} An object with product ID
+ *     keys that each have a `ShoppingConfigDataDef` value
  */
-export function getShoppingConfig(element) {
-  return getElementConfig(element).then((config) => {
-    //TODO(#36412): Add call to validate config here.
-    return keyByProductTagId(config);
+export function getShoppingConfig(shoppingAttachmentEl) {
+  return getElementConfig(shoppingAttachmentEl).then((config) => {
+    const pageId = closestAncestorElementBySelector(
+      shoppingAttachmentEl,
+      'amp-story-page'
+    ).id;
+
+    const allItems = config['items'];
+    const validItems = allItems.filter((item, i) => {
+      const productTitle = item?.productTitle ?? '';
+      return validateConfig(
+        item,
+        `#${pageId} items[${i}] ${productTitle}`.trim()
+      );
+    });
+
+    return keyByProductId(validItems);
   });
 }
 
 /**
- * @param {!ShoppingConfigResponseDef} config
+ * Takes an array of product configs and returns a map of product IDs to
+ * product configs.
+ * @param {!ShoppingConfigResponseDef} configItems
```

Comment:
This type is wrong now.
```suggestion
 * @param {ShoppingConfigDataDef[]} configItems
```